components/ArticleCard.jsx

```diff
@@ -1,10 +1,20 @@
+import { useNavigate, Link } from "react-router-dom";
+
 export const ArticleCard = ({ article }) => {
+  const navigate = useNavigate();
+  const handleNavigate = (e) => {
```

Comment:
This is not the appropriate way to handle navigating the user. This should be a link. Links are for manual navigation and are rendered as a tags for screen readers. The method you have used here is not accessible.